website/src/blogs/litmus-journey-of-a-storage-e2e-and-chaos-framework-for-kubernetes.md

```diff
@@ -0,0 +1,96 @@
+
+One of the important enablers for the adoption of any given test (suite) into a CI framework is the result mechanism to determine success/failure. During the said period, Litmus moved over from extracting results from test container logs to storing this information in a Kubernetes static custom resource (*“litmusresult”, which is updated by the test code at the beginning & end of test execution via a JSON merge patch operation*), with the resource name dynamically generated based on user tags. This resource also holds other test metadata such as *appType*, *testType*, *testStatus *etc., apart from the actual *verdict *(result). CI jobs that implement/run the litmusbooks can be configured to derive job status from these CRs.
+
+![cr-structure](/pubic/images/blog/cr-structure.png)A jinja template describing the result CR structure
+
+## Other notable enhancements
+
+The project also saw some other organic changes which have resulted in its faster adoption into CI systems & increased efficiency of the tests. They have been briefly described below:
+
+**Infra Setup Playbooks**: Litmus now provides multiple ansible playbooks to aid quick and easy creation of Kubernetes clusters on different popular cloud platforms (GCP, AWS, GKE, AKS, EKS, Packet Cloud, On-premise vSphere VMs, OpenShift-on-AWS, OpenShift-on-vSphere) with minimal config inputs & also litmusbooks to setup the OpenEBS control plane on a K8s cluster.
+
+**Simplifying Litmus Setup**: The litmus pre-requisites are taken care of by a simple interactive bash script that sets up the RBAC, CRDs & config file updates.
+
+**K8s-Specific Enhancements**: These include small changes to make the litmusbooks more robust, such as:-
+
+- Usage of generateName API in the litmus jobs to ensure repeated execution doesn’t cause failures
+- Maximum Usage of labels & JSONPATH expressions in the kubectl commands used in the test code instead of bash utilities to guard against K8s cluster version & storage resource changes.
+- Increased use of pod ENV variables (user-specific & downward API) to control test inputs in a K8s-native way rather than pass them in a test-code specific form (ansible extra-vars)
+
+**Increased Modularization of Test Code**: Most of the test code is reused across the litmusbooks, due to which reusable tasks are managed as separate utils. These are further classified as *common*, *chaos* & *app/storage lifecycle*modules, with well-defined interfaces.
+
+**Litmus CI**: The Litmus (and test-tools) project now consists of its own Travis-based CI which builds respective runner images post-Ansible/Python lint/syntax check operations.
+
+**Optimized Test Images**: The litmus runner (go, ansible) as well the test facilitators images have been optimized with respect to the number of layers, size and binary versions used.
+
+## Case Study: OpenEBS CI Powered By Litmus
+![](/content/images/2020/01/openebs-ci-powered-by-litmus.png)OpenEBS CI Powered By Litmus
```

Comment:
Please add this image to images/blog folder. I see this in a couple of places. Please move those images to this folder too

Reply:
Also, add space between image and text. If it is a figure caption, then we can display this in this format (****Figure caption***) below image without a line space